sdk/template/Azure.Template/src/Generated/MultiVersionClient.cs

```diff
@@ -0,0 +1,237 @@
+            if (_serviceVersion < MultiVersionClientOptions.ServiceVersion.V2022_02_02)
+            {
+                throw new NotSupportedException($"Operation not available in service version {_serviceVersion}");
+            }
+
+            using var scope = ClientDiagnostics.CreateScope("MultiVersionClient.ServiceV2Operation");
+            scope.Start();
+            try
+            {
+                using HttpMessage message = CreateServiceV2OperationRequest(context);
+                return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <param name="context"> The request context, which can override default behaviors of the client pipeline on a per-call basis. </param>
+        /// <exception cref="RequestFailedException"> Service returned a non-success status code. </exception>
+        /// <returns> The response returned from the service. </returns>
+        /// <example>
+        /// This sample shows how to call ServiceV2Operation and parse the result.
+        /// <code><![CDATA[
+        /// var credential = new DefaultAzureCredential();
+        /// var endpoint = new Uri("<https://my-service.azure.com>");
+        /// var client = new MultiVersionClient(endpoint, credential);
+        /// 
+        /// Response response = client.ServiceV2Operation();
+        /// 
+        /// JsonElement result = JsonDocument.Parse(response.ContentStream).RootElement;
+        /// Console.WriteLine(result.ToString());
+        /// ]]></code>
+        /// </example>
+        public virtual Response ServiceV2Operation(RequestContext context = null)
+        {
+            if (_serviceVersion < MultiVersionClientOptions.ServiceVersion.V2022_02_02)
+            {
+                throw new NotSupportedException($"Operation not available in service version {_serviceVersion}");
+            }
+
+            using var scope = ClientDiagnostics.CreateScope("MultiVersionClient.ServiceV2Operation");
+            scope.Start();
+            try
+            {
+                using HttpMessage message = CreateServiceV2OperationRequest(context);
+                return _pipeline.ProcessMessage(message, context);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        internal HttpMessage CreateServiceV1OperationRequest(RequestContext context)
+        {
+            var message = _pipeline.CreateMessage(context, ResponseClassifier200);
+            var request = message.Request;
+            request.Method = RequestMethod.Get;
+            var uri = new RawRequestUriBuilder();
+            uri.Reset(_endpoint);
+            uri.AppendPath("/ops/v1op", false);
+            uri.AppendQuery("api-version", _apiVersion, true);
```

Comment:
Validate that api-version is added to request (fail generator if operation doesn't support api-version parameter)